sklearn/utils/testing.py

```diff
@@ -625,8 +625,9 @@ def run_test(*args, **kwargs):
             import matplotlib
             matplotlib.use('Agg', warn=False)
             # this fails if no $DISPLAY specified
-            matplotlib.pylab.figure()
-        except:
+            import matplotlib.pyplot as plt
+            plt.figure()
+        except ImportError:
```

Comment:
I'd like to merge the fix independent of this, though. Current master tests are just broken :-/